rct229/rule_engine/partial_rule_definition.py

```diff
@@ -0,0 +1,99 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.utils.json_utils import slash_prefix_guarantee
+
+
+class PartialRuleDefinition(RuleDefinitionBase):
+    def __init__(
+        self,
+        rmrs_used,
+        id=None,
+        description=None,
+        ruleset_section_title=None,
+        standard_section=None,
+        is_primary_rule=False,
+        rmr_context="",
+        required_fields=None,
+        must_match_by_ids=[],
```

Comment:
I don't think `must_match_by_ids` is used anywhere in the rule engine.  I think it should be removed here and from rule_base.py.  I don't recall when or why this was added.